### ISSUE TITLE
[android_intend_plus] how to launch componets in com.google.android.googlequicksearchbox

### ISSUE THREAD

**Issue by Shiba-Kar** [fluttercommunity/plus_plugins#548]
**I want to access the google lens activity  which is present in the google Quick Search bar _(com.google.android.googlequicksearchbox)_**

#503 shows how can you launch if the google lens APP is install . Searching a way to launch the inbuild google lens activity which is shipped with google quick search.

So, coded the below no luck.

```
if (platform.isAndroid) {
  final AndroidIntent intent = AndroidIntent(
      package:"com.google.android.googlequicksearchbox",
      component:"com.google.android.apps.lens.MainActivity"
  );
  await intent.launch();
}
```

Am I missing something here !!

**Comment by alexbatalov** [comment]
**tl;dr** You can't

---

You can open the search itself, the component name is `com.google.android.googlequicksearchbox.SearchActivity`, which is pretty useless because you can't get any result:
```dart
import 'package:android_intent_plus/android_intent.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: ElevatedButton(
            child: const Text('Open Google Search'),
            onPressed: () {
              const intent = AndroidIntent(
                package: 'com.google.android.googlequicksearchbox',
                componentName:
                    'com.google.android.googlequicksearchbox.SearchActivity',
              );
              intent.launch();
            },
          ),
        ),
      ),
    );
  }
}
```

There are many other activities in the package, including `com.google.android.apps.lens.MainActivity` which is an alias for `com.google.android.apps.search.lens.LensExportedActivity`. If you try to use it you'll notice a blink for a second - that was Lens activity opened and closed immediately. It's intended usage is thru `startActivityForResult`, so that Lens can return result to the calling application. Because `android_intent_plus` does not use `startActivityForResult`, the `getCallingPackage` returns `null`, so when Lens activity starts it knows that there's no calling application and quits. Even if you write your own native code (like I did) to `startActivityForResult`, there is another safe-guard - Google maintains a private list of package ids, that are allowed to open Lens. Because your package is obviously not on this list you'll see `Package com.flutter.demo not on LENS_EXPORTED_ACTIVITY_ALLOWLIST` in logs. There are might be other checks, I didn't investigate further.

<details>
<summary> LENS_EXPORTED_ACTIVITY_ALLOWLIST</summary>

```
com.google.android.inputmethod.latin
com.google.android.inputmethod.latin.dev
com.google.android.inputmethod.latin.canary
com.google.android.apps.gmm
com.google.android.apps.gmm.dev
com.google.android.apps.gmm.dogfood
com.google.android.apps.gmm.fishfood
com.google.android.apps.gmm.qp
com.google.android.apps.maps
com.google.android.apps.photos
com.google.android.apps.translate
com.google.android.apps.translatedebug
com.google.android.youtube
com.google.android.apps.search.lingocamera
com.google.android.apps.search.niu.dev
com.google.android.googlequicksearchbox
com.google.android.apps.gsa.staticplugins.opa
com.google.android.apps.chrome
com.google.android.apps.lens.demo.assist
com.chrome.dev
com.chrome.canary
com.chrome.beta
com.android.chrome
com.google.ar.lens
com.google.ar.lens.dev
```
</details>
